python/cudf_polars/cudf_polars/dsl/ir.py

```diff
@@ -933,10 +933,10 @@ class Union(IR):
     """Optional slice to apply after concatenation."""
 
     def __post_init__(self) -> None:
-        """Validated preconditions."""
+        """Validate preconditions."""
         schema = self.dfs[0].schema
         if not all(s.schema == schema for s in self.dfs[1:]):
-            raise ValueError("Schema mismatch")
```

Comment:
Why the change here?

Is the idea to give user the polars version of the error message.

Reply:
The promise of `translate_ir` is that it will fail, with `NotImplementedError`, if we can't do the thing. This `ValueError` exception broke that promise. Arguably this is just a stop-gap for polars not validating the schema early. I'm not trying to match polars exceptions one-for-one.